staging/src/k8s.io/client-go/tools/clientcmd/config.go

```diff
@@ -58,6 +58,13 @@ type PathOptions struct {
 	LoadingRules *ClientConfigLoadingRules
 }
 
+var (
+	// UseConfigLock ensures that access to kubeconfig files is being guarded by a lock file.
+	// This variable is intentionaly made public so other consumers of this library
+	// can modify its default behavior.
+	UseConfigLock = true
```

Comment:
make the name and godoc for this specific to ModifyConfig (e.g. `UseModifyConfigLock`) and note that it is not threadsafe

Reply:
For exposing this in kubectl, I would expect a `--nolock` option to the modifying config commands. That seems like it would work better plumbed into ModifyConfig than as a global var

Reply:
Yeah, I was thinking of a flag, rather than env var.